moveit_ros/hybrid_planning/global_planner/global_planner_component/src/global_planner_component.cpp

```diff
@@ -0,0 +1,160 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, PickNik Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+ */
+
+#include <moveit/global_planner/global_planner_component.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/robot_state/conversions.h>
+#include <moveit_msgs/msg/move_it_error_codes.h>
+
+#include <chrono>
+#include <thread>
+
+namespace moveit_hybrid_planning
+{
+using namespace std::chrono_literals;
+const rclcpp::Logger LOGGER = rclcpp::get_logger("global_planner_component");
+const std::string UNDEFINED = "<undefined>";
+
+GlobalPlannerComponent::GlobalPlannerComponent(const rclcpp::NodeOptions& options)
+  : Node("global_planner_component", options)
+{
+  // Initialize global planner after construction
+  // TODO(sjahr) Remove once life cycle component nodes are available
+  timer_ = this->create_wall_timer(1ms, [this]() {
+    if (initialized_)
+    {
+      timer_->cancel();
+    }
+    else
+    {
+      initialized_ = this->init();
+      if (!initialized_)
+      {
+        timer_->cancel();
+        throw std::runtime_error("Failed to initialize global planner");
+      }
+    }
+  });
+}
+
+bool GlobalPlannerComponent::init()
+{
+  auto node_ptr = shared_from_this();
```

Comment:
This is only used in one place (L124). Suggest moving closer to where it's used (L123).